add bundleReference to the claim specification

adamreese opened the pull request:
ref: [New features in Claims specification require changes here](https://github.com/cnabio/libcnab-rust/issues/30)
https://github.com/cnabio/cnab-spec/pull/299

glyn commented on src/claim.rs:
```diff
@@ -26,6 +26,10 @@ pub struct Claim {
     pub result: Response,
     /// A ulid to track the revision
     pub revision: String,
+    /// A canonical reference to the bundle used in the last action.
```
```suggestion
    /// An optional canonical reference to the bundle used in the last action.
```

radu-matei replied:
If this field is optional, could we represent it as `Option<String>` instead?

glyn replied:
It either already is or it has already been updated to be so.

radu-matei replied:
Oh, you're right.

glyn commented on src/claim.rs:
```diff
@@ -26,6 +26,10 @@ pub struct Claim {
     /// A ulid to track the revision
     pub revision: String,
+    /// A canonical reference to the bundle used in the last action.
+    /// This bundle SHOULD use a SHA to identify the specific version of the referenced bundle.
```
```suggestion
    /// When provided, the bundle reference SHOULD be digested to identify a specific version of the referenced bundle.
```

technosophos reviewed (approved):
Yeah, with Glyn's minor doc changes, this is totally LGTM. Thanks!

glyn reviewed (changes requested):
I think my remaining suggestion should be applied for consistency with the spec.

adamreese committed:
add bundleReference to the claim specification

https://github.com/cnabio/libcnab-rust/issues/30
https://github.com/cnabio/cnab-spec/pull/299

Signed-off-by: Adam Reese <adam@reese.io>

radu-matei commented:
I think this needs a rebase, then we can merge.
Thanks!

technosophos committed:
Merge branch 'master' into feat/30-sync-to-spec
